libs/barista-components/filter-field/src/filter-field-util.ts

```diff
@@ -78,24 +78,22 @@ export function filterAutocompleteDef(
 export function filterFreeTextDef(
   def: DtNodeDef,
   filterText?: string,
-): DtNodeDef | null {
+): DtNodeDef {
   const suggestions = def.freeText!.suggestions
     ? def.freeText!.suggestions.filter((option) =>
         filterOptionDef(option, new Set(), filterText),
       )
     : [];
 
-  return def.freeText!.async || suggestions.length
```

Comment:
a freeText doesn't always have to be async or have suggestions

Reply:
I've tested the async behavior and it still works for freetext with these changes applied.